service/frontend/middleware/global.go

```diff
@@ -12,12 +13,16 @@ func SetupGlobalMiddleware(handler http.Handler) http.Handler {
 	next = middleware.Logger(next)
 	next = middleware.Recoverer(next)
 
+	if authToken != nil {
+		next = TokenAuth("bearer", authToken.Token)(next)
+		basicAuth = nil
+	}
+
 	if basicAuth != nil {
 		next = middleware.BasicAuth(
 			"restricted", map[string]string{basicAuth.Username: basicAuth.Password},
 		)(next)
```

Comment:
I think we can add a meta middleware (like `skipper` middleware) to skip a request with api token, which allows us to check the http header and skip if the request holds a API token in the `Authentication` header. But it requires additional work so please don't mind for this PR.